File: code/Ivar.m
#import <Foundation/Foundation.h>
#import <objc/runtime.h>

@interface MyClass : NSObject

@property (nonatomic, retain) NSString* foo;

@end

#pragma mark -

@implementation MyClass

-(instancetype)init {
    self = [super init];
    if (self) {
        self.foo = @"bar";
    }
    return self;
}

@end

#pragma mark -

int main(int argc, char *argv[])
{
    @autoreleasepool {
        MyClass* object = [MyClass new];
        Ivar ivar = class_getInstanceVariable(object.class, "_foo");
        id value = object_getIvar(object, ivar);
        NSLog(@"%@", value);
        return 0;
    }
}

// clang Ivar.m -framework Foundation

File: code/Ivar.swift
#!/usr/bin/xcrun swift

class MySwiftClass /*: NSObject*/ {
    var foo = "bar";

    /*override*/ init() {
    }
}

import Foundation
import ObjectiveC.runtime

var ivar = class_getInstanceVariable(MySwiftClass().dynamicType, "foo")

print(NSString(CString: ivar_getName(ivar), encoding: NSUTF8StringEncoding)!)

var value : AnyObject = object_getIvar(MySwiftClass(), ivar)!

print(value)
